source/core/events/ut_event_manager.pkb

```diff
@@ -21,15 +21,42 @@ create or replace package body ut_event_manager  as
   type t_listener_numbers is table of boolean index by t_listener_number;
   type t_events_listeners is table of t_listener_numbers index by t_event_name;
 
-  g_event_listeners_index t_events_listeners;
-  g_listeners             t_listeners;
+  type t_event_manager is record (
+    event_listener_index t_events_listeners,
+    listeners            t_listeners
+  );
+  type t_event_managers is table of t_event_manager;
+
+  g_event_listeners_index    t_events_listeners;
+  g_listeners                t_listeners;
+  g_suspended_event_managers t_event_managers;
 
   procedure initialize is
   begin
+    if g_listeners is not null and g_listeners.count > 0 then
+       if g_suspended_event_managers is null then
+         g_suspended_event_managers := t_event_managers();
+       end if;
+       g_suspended_event_managers.extend;
+       g_suspended_event_managers(g_suspended_event_managers.count).event_listener_index := g_event_listeners_index;
+       g_suspended_event_managers(g_suspended_event_managers.count).listeners            := g_listeners;
+    end if;
     g_event_listeners_index.delete;
     g_listeners := t_listeners();
   end;
 
+  procedure dispose_listeners is
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Avoid procedures without parameters. ([plsql:ProcedureWithoutParametersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AProcedureWithoutParametersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=809&issues=AWgIpEiuSi5GWQS9jhj6&open=AWgIpEiuSi5GWQS9jhj6)

[ISSUE_KEY]: AWgIpEiuSi5GWQS9jhj6